### PR TITLE
Shivansh | Test | Replaced checkboxes with toggle button

### PR DESCRIPTION
**Pull request by shivanshRawat14**
https://app.clickup.com/t/2839xxp

### CODE REVIEW

File: apps/web-giddh/src/app/inventory/components/add-stock-components/inventory.addstock.component.ts [hunk 1]
```diff
@@ -758,6 +758,10 @@ export class InventoryAddStockComponent implements OnInit, AfterViewInit, OnDest
                 this.addStockForm?.patchValue({ isFsStock: false });
             }
         }
+
+        this.addStockForm.controls['enablePurchase']?.patchValue(false);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/inventory/components/add-stock-components/inventory.addstock.component.ts]: remove this

File: apps/web-giddh/src/app/invoice/invoice.module.ts [hunk 2]
```diff
@@ -4,14 +4,15 @@ import { ProformaInvoiceModule } from '../proforma-invoice/proforma-invoice.modu
...
+
 @NgModule({
     declarations: [],
     imports: [
         InvoiceRoutingModule,
         ProformaInvoiceModule,
         DigitsOnlyModule,
         PurchaseModule,
-        SharedModule
+        SharedModule,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/invoice.module.ts]: revert changes of this file

File: apps/web-giddh/src/app/invoice/settings/invoice.settings.component.ts [hunk 3]
```diff
@@ -148,6 +159,8 @@ export class InvoiceSettingComponent implements OnInit, OnDestroy {
                 this.invoiceSetting = cloneDeep(setting.invoiceSettings);
                 this.proformaSetting = cloneDeep(setting.proformaSettings);
                 this.invoiceSetting.autoPaid = this.invoiceSetting.autoPaid === 'runtime';
+                
+
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/settings/invoice.settings.component.ts]: remove this space

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html [hunk 4]
```diff
@@ -252,8 +252,20 @@ <h3 mat-dialog-title class="modal-title bg unbold" id="modal-title">
                             <div class="row">
                                 <div class="col-6 col-md-6">
                                     <div class="form-group">
-                                        <mat-checkbox color="primary" formControlName="cancelledEntries">
-                                        </mat-checkbox>
+                                        <!-- <mat-checkbox color="primary" formControlName="cancelledEntries">
+                                        </mat-checkbox> -->
+                                        <mat-slide-toggle color="primary" formControlName="cancelledEntries">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html:257]: test this

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.html [hunk 5]
```diff
@@ -90,21 +90,33 @@ <h3 class="unbold" mat-dialog-title>{{ localeData?.page_heading }}</h3>
...
                     <label class="w125">{{ commonLocaleData?.app_invoice_no }}:</label>
                     <label class="radio-inline p-0">
-                        <mat-checkbox
+                        <!-- <mat-checkbox
                             color="primary"
                             name="withInvoiceNumber"
                             [(ngModel)]="withInvoiceNumber"
-                        ></mat-checkbox>
+                        ></mat-checkbox> -->
+                        <mat-slide-toggle color="primary" name="withInvoiceNumber" [(ngModel)]="withInvoiceNumber">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.html:98]: test this

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.html [hunk 6]
```diff
@@ -90,21 +90,33 @@ <h3 class="unbold" mat-dialog-title>{{ localeData?.page_heading }}</h3>
...
                     <label class="w125">{{ localeData?.cr_dr }}</label>
                     <label class="radio-inline p-0">
-                        <mat-checkbox
+                        <!-- <mat-checkbox
                             color="primary"
                             name="balanceTypeAsSign"
                             [(ngModel)]="balanceTypeAsSign"
-                        ></mat-checkbox>
+                        ></mat-checkbox> -->
+                        <mat-slide-toggle color="primary" name="balanceTypeAsSign" [(ngModel)]="balanceTypeAsSign">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.html:114]: test this

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html [hunk 7]
```diff
@@ -212,11 +212,22 @@ <h3 class="font-20 bold mb-2">{{ localeData?.collection?.razorpay_form_heading }
...
                                     <br />
-                                    <input
+                                    <!-- <input
                                         type="checkbox"
                                         [(ngModel)]="razorPayObj.autoCapturePayment"
                                         name="autoCapturePayment"
-                                    />
+                                    /> -->
+                                    <ngx-switch
+                                        [status]="razorPayObj.autoCapturePayment"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.html]: change this

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.html [hunk 8]
```diff
@@ -262,14 +262,32 @@ <h3>{{ commonLocaleData?.app_address }} {{ i + 1 }}</h3>
...
                                                     name="defaultGst_{{ i }}"
                                                     formControlName="isDefault"
                                                     (change)="isDefaultAddressSelected(gst.get('isDefault').value, i)"
                                                     [tooltip]="localeData?.address_tooltip"
                                                     placement="left"
-                                                />
+                                                /> -->
+                                                <span [tooltip]="localeData?.address_tooltip" placement="left">
+                                                    <ngx-switch
+                                                        [(status)]="
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.html:275]: test this

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 9]
```diff
@@ -403,6 +403,7 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
         });
         return gstFields;
     }
+    public varTest = this.initialGstDetailsForm;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]: remove if unused

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html [hunk 10]
```diff
@@ -245,14 +245,32 @@
...
                                                     name="defaultGst_{{ i }}"
                                                     formControlName="isDefault"
                                                     (change)="isDefaultAddressSelected(gst.get('isDefault').value, i)"
                                                     [tooltip]="localeData?.address_tooltip"
                                                     placement="left"
-                                                />
+                                                /> -->
+                                                <span [tooltip]="localeData?.address_tooltip" placement="left">
+                                                    <ngx-switch
+                                                        [status]="
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html]: test this

File: apps/web-giddh/src/app/userDetails/userDetails.component.ts [hunk 11]
```diff
@@ -19,6 +19,7 @@ import { BsModalRef } from 'ngx-bootstrap/modal';
 import { GeneralActions } from '../actions/general/general.actions';
 import { API_POSTMAN_DOC_URL } from '../app.constant';
 import { BreakpointObserver } from '@angular/cdk/layout';
+import { fakeAsync } from '@angular/core/testing';
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/userDetails/userDetails.component.ts]: remove this if unused

File: package.json [hunk 12]
```diff
@@ -70,7 +70,7 @@
   },
   "private": true,
   "dependencies": {
-    "@angular/animations": "12.1.1",
+    "@angular/animations": "^12.1.1",
```
Comment (ravinderwalkover) [inline package.json]: revert this

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/inventory/components/add-stock-components/inventory.addstock.component.ts (hunk 1 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/inventory/components/add-stock-components/inventory.addstock.component.ts]: removed

File: apps/web-giddh/src/app/invoice/invoice.module.ts (hunk 2 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/invoice/invoice.module.ts]: reverted

File: apps/web-giddh/src/app/invoice/settings/invoice.settings.component.ts (hunk 3 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/invoice/settings/invoice.settings.component.ts]: removed

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html (hunk 7 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/settings/integration/setting.integration.component.html]: changed

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts (hunk 9 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]: removed

File: apps/web-giddh/src/app/userDetails/userDetails.component.ts (hunk 11 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/userDetails/userDetails.component.ts]: removed

File: package.json (hunk 12 above)
Comment (shivanshRawat14) [reply package.json]: reverted

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.html (hunk 5 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.html:98]: tested

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.html (hunk 6 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.html:114]: tested

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html (hunk 4 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html:257]: tested

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.html (hunk 8 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.html:275]: tested

**Review by ravinderwalkover — APPROVED** [review]

File: apps/web-giddh/src/app/app.constant.ts [hunk 13]
```diff
@@ -441,3 +441,15 @@ export const JOURNAL_VOUCHER_ALLOWED_DOMAINS = [
...
     'whozzat.com',
 ];
+
+/**
+ * Enum for switching toggle button On and Off and changing its size
+ *
+ * @export
+ * @enum {string}
+ */
+export enum BOOTSTRAP_TOGGLE_SWITCH {
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/app.constant.ts]: Enums are always named in CamelCase
```suggestion
export enum BootstrapToggleSwitch {
```

File: apps/web-giddh/src/app/invoice/settings/invoice.settings.component.ts [hunk 14]
```diff
@@ -85,6 +90,17 @@ export class InvoiceSettingComponent implements OnInit, OnDestroy {
         private router: Router,
         private generalService: GeneralService
     ) {
+        this.invoiceSetting.showSeal = false;
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/invoice/settings/invoice.settings.component.ts]: Can we assign these default values in the class's constructor itself called at https://github.com/Walkover-Web-Solution/Giddh-New-Angular4-App/blob/3a08674c9ccb75377f2dafba692a694f23730db1/apps/web-giddh/src/app/invoice/settings/invoice.settings.component.ts#L30

File: apps/web-giddh/src/app/purchase/purchase-setting/purchase-setting.component.ts [hunk 15]
```diff
@@ -72,6 +75,10 @@ export class PurchaseSettingComponent implements OnInit, OnDestroy {
 
         this.store.dispatch(this.settingsIntegrationActions.GetGmailIntegrationStatus());
         this.initSettings();
+        this.invoiceSettings.purchaseBillSettings.sendThroughGmail = false;
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/purchase/purchase-setting/purchase-setting.component.ts]: Can we assign these default values at line https://github.com/Walkover-Web-Solution/Giddh-New-Angular4-App/blob/3a08674c9ccb75377f2dafba692a694f23730db1/apps/web-giddh/src/app/purchase/purchase-setting/purchase-setting.component.ts#L25

**Review by the-prateek-mishra — CHANGES_REQUESTED** [review]

**Review by the-prateek-mishra — APPROVED** [review]

**Pull request merged**